stew/results.nim

```diff
@@ -846,6 +852,8 @@ func flatten*[T, E](self: Result[Result[T, E], E]): Result[T, E] =
     else:
       err(Result[T, E], self.error)
 
+{.push effectsOf: callback.}
```

Comment:
this goes in the function declaration itself so that it's visible - it's critical information for the claler - that said, it also doesn't exist/work on 1.2 so we can't use it there yet

Reply:
https://github.com/status-im/nim-stew/pull/146/commits/738b75f0414920bb8351fe63b6267ad5d9e73ff2